src/django_components/component.py

```diff
@@ -303,135 +320,195 @@ def get_context_data(self):
 
         As the `{{ data.hello }}` is taken from the "provider".
         """
-        comp_name = self.registered_name or self.__class__.__name__
         if self._context is None:
             raise RuntimeError(
-                f"Method 'inject()' of component '{comp_name}' was called outside of 'get_context_data()'"
+                f"Method 'inject()' of component '{self.name}' was called outside of 'get_context_data()'"
             )
 
-        return get_injected_context_var(comp_name, self._context, key, default)
+        return get_injected_context_var(self.name, self._context, key, default)
+
+    @classmethod
+    def render_to_response(
+        cls,
+        context: Union[Dict[str, Any], Context] = None,
+        slots: Optional[Mapping[SlotName, SlotContent]] = None,
+        escape_slots_content: bool = True,
+        args: Optional[Union[List, Tuple]] = None,
+        kwargs: Optional[Dict[str, Any]] = None,
+        *response_args: Any,
+        **response_kwargs: Any,
+    ) -> HttpResponse:
+        """
+        This is the interface for the `django.views.View` class which allows us to
+        use components as Django views with `component.as_view()`.
+
+        Example:
+        ```py
+        MyComponent.render_to_response(
+            args=[1, "two", {}],
+            kwargs={
+                "key"=123,
+            },
+            slots={
+                header='STATIC TEXT HERE',
+            },
+            escape_slots_content=False,
+        )
+        ```
+        """
+        content = cls.render(
+            args=args,
+            kwargs=kwargs,
+            context=context,
+            slots=slots,
+            escape_slots_content=escape_slots_content,
+        )
+        return cls.response_class(content, *response_args, **response_kwargs)
+
+    @classmethod
+    def render(
+        cls,
+        context: Optional[Union[Dict[str, Any], Context]] = None,
+        args: Optional[Union[List, Tuple]] = None,
+        kwargs: Optional[Dict[str, Any]] = None,
+        slots: Optional[Mapping[SlotName, SlotContent]] = None,
+        escape_slots_content: bool = True,
+    ) -> str:
+        """
+        Example:
+        ```py
+        MyComponent.render(
+            args=[1, "two", {}],
+            kwargs={
+                "key"=123,
+            },
+            slots={
+                header='STATIC TEXT HERE',
+            },
+            escape_slots_content=False,
+        )
+        ```
+        """
+        comp = cls()
+        return comp._render(context, args, kwargs, slots, escape_slots_content)
 
-    def render_from_input(
+    def _render(
```

Comment:
Merged `render` and `render_from_input` into `_render`. 

Note the difference:
- `render` is public method and class method
- `_render` is private method and instance method.

`render` (public class method) internally instantiates Component class and calls `_render`.